src/uct/sm/mm/mm_ep.h

```diff
@@ -13,10 +13,13 @@
 #include <uct/tl/tl_log.h>
 
 
-typedef struct uct_mm_ep {
-    uct_base_ep_t      super;
-    struct uct_mm_ep *next;
-} uct_mm_ep_t;
+struct uct_mm_ep {
+    uct_base_ep_t           super;
+
+    /* Remote peer */
+    uct_mm_fifo_ctl_t       *fifo_ctl;  /* pointer to the beginning of the destination's receive fifo */
+    void                    *fifo;      /* fifo elements (destination's receive fifo) */
```

Comment:
BTW get_elem_to_write() can still return a non-volatile pointer, thus asserting that the element is in complete ownership of the current process, and is not volatile